tests/tarp_test.py

```diff
@@ -141,12 +141,46 @@ def test_run_tarp_correct(distance, z_score_theta, accurate_samples):
         num_bins=30,
     )
 
-    assert allclose((ecp - alpha).abs().max(), Tensor([0.0]), atol=1e-1)
+    assert allclose((ecp - alpha).abs().max(), zeros((1,)), atol=1e-1)
     assert (
         ecp - alpha
     ).abs().sum() < 1.0  # integral of residuals should vanish, fig.2 in paper
 
 
+@pytest.mark.gpu
+@pytest.mark.parametrize("distance", (l1, l2))
+@pytest.mark.parametrize("z_score_theta", (True, False))
```

Comment:
do we need to iterate on these 4 conditions again for the GPU case as well? That this runs correctly should have been tested above and is independent of the device? 
If so, let's remove it to avoid test repetitions.